include/pybind11/numpy.h

```diff
@@ -767,6 +769,9 @@ template <typename T, int ExtraFlags = array::forcecast> class array_t : public
     explicit array_t(ShapeContainer shape, const T *ptr = nullptr, handle base = handle())
         : array(std::move(shape), ptr, base) { }
 
+    explicit array_t(size_t count, const T *ptr = nullptr, handle base = handle())
```

Comment:
+1 for changing the `pybind11_fail` to `error_already_set` in the main constructor.